login/service.go

```diff
@@ -162,18 +161,29 @@ func (keycloak *KeycloakOAuthProvider) Perform(ctx *app.AuthorizeLoginContext, a
 	}
 
 	// First time access, redirect to oauth provider
+	redirect := ctx.Redirect
+	referrer := ctx.RequestData.Header.Get("Referer")
```

Comment:
Absolutely. We must whitelist redirects. There is a separate issue for that https://github.com/almighty/almighty-core/issues/892
But this PR doesn't make it worse, or it actually does?